src/app/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import defaultAvatar from './defaultAvatar.svg';
+import classes from './Profile.module.css';
+
+type Profile = {
+  username: string | null;
+  displayName: string | null;
+  avatar: string | null;
+};
+
+function openDialog() {
+  overwolf.profile.openLoginDialog();
+}
+
+function Profile(): JSX.Element {
+  const defaultProfile: Profile = {
+    username: 'Game-Ticker',
+    displayName: 'Game-Ticker',
+    avatar: null,
+  };
+
+  const [profile, setProfile] = useState(defaultProfile);
+  const [profileError, setProfileError] = useState<string | null>(null);
+  const [loginDialog, setLoginDialog] = useState(false);
+
+  useEffect(() => {
+    try {
+      overwolf.profile.getCurrentUser((result) => {
```

Comment:
You could think about a promised version of `getCurrentUser`

Reply:
done, be aware of ugly code